src/common/Exporter.js

```diff
@@ -0,0 +1,80 @@
+"use strict";
+(function (root, factory) {
+    if (typeof define === "function" && define.amd) {
+        define([], factory);
+    } else {
+        root.common_Exporter = factory();
+    }
+}(this, function () {
+    function Exporter(widget) {
+        this._widget = widget;
+    }
+
+    Exporter.prototype.formatData = function (_) {
+        var formattedData;
+        switch(_) {
+            case "TSV":
+                formattedData = this._widget._db.tsv();
+                break;
+            case "JSON":
+                formattedData = this._widget._db.json();
+                break;
+            default:
+                formattedData = this._widget._db.csv();
+        }
+        return formattedData;
+    };
+
+
+    Exporter.prototype.downloadData = function (format) {
```

Comment:
Can we move this into Utility?   We will pass in the pre-formatted data and a type.